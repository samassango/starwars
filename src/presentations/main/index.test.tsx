import React from 'react';
import { render, screen } from '@testing-library/react';
import { shallow } from 'enzyme';
import Main from './index';

const mockDispatch = jest.fn();

jest.mock('react-redux', () => ({
    ...jest.requireActual('react-redux'),
    useSelector: jest.fn(),
    useDispatch: () => mockDispatch,
}));

jest.mock('react-router-dom',()=>({
...jest.requireActual('react-router-dom'),
useNavigate:()=>jest.fn()
}));

test('renders main component', () => {
  const wrapper = shallow(<Main />);
  expect(wrapper).toBeTruthy();
});

test('renders main component', () => {
    const wrapper = shallow(<Main />);
    expect(wrapper).toBeTruthy();
  });

  test('component should contain "ROW"',()=>{
    const wrapper = shallow(<Main />);
    expect(wrapper.find('.justify-content-md-center')).toBeTruthy();
});

test('test component snapshot',()=>{
    const {container, getByText} = render(<Main />);
    expect(container.firstChild).toMatchSnapshot()
})
