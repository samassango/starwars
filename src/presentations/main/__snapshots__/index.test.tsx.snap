// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test component snapshot 1`] = `
<div
  class="container"
>
  <div
    class="justify-content-md-center row"
  >
    <div
      class="row"
    >
      <div
        class="col"
      >
        <div
          class="fade alert alert-success show"
          role="alert"
        >
          <h3>
            Star Wars Dashboard
          </h3>
        </div>
      </div>
    </div>
    <table
      class="table table-striped table-bordered table-hover"
    >
      <thead>
        <tr>
          <th>
            #
          </th>
          <th>
            Fullname
          </th>
          <th>
            Height
          </th>
          <th>
            Gender
          </th>
          <th>
            Mass
          </th>
          <th>
            Homeworld
          </th>
          <th>
            Action
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
    <div
      class="row"
    >
      <div
        class="col-md-6 offset-md-5"
      >
        <ul
          class="pagination"
        >
          <li
            class="page-item active"
          >
            <span
              class="page-link"
            >
              1
              <span
                class="visually-hidden"
              >
                (current)
              </span>
            </span>
          </li>
          <li
            class="page-item"
          >
            <a
              class="page-link"
              role="button"
              tabindex="0"
            >
              2
            </a>
          </li>
          <li
            class="page-item"
          >
            <a
              class="page-link"
              role="button"
              tabindex="0"
            >
              3
            </a>
          </li>
          <li
            class="page-item"
          >
            <a
              class="page-link"
              role="button"
              tabindex="0"
            >
              4
            </a>
          </li>
          <li
            class="page-item"
          >
            <a
              class="page-link"
              role="button"
              tabindex="0"
            >
              5
            </a>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
